apps/client/public/app.js

```
@@ -0,0 +1,143 @@
+// one place that needs its own user-visible failure path.
+async function postAction(action, fields) {
+    try {
+        const res = await fetch(`/s/${sessionId}/actions/${action}`, {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify(fields)
+        });
+        return await res.json();
+    } catch (error) {
+        console.error(`${action} failed:`, error);
+        alert(`${action} failed: ${error.message}`);
```

ESLint

[error] 'alert' is not defined.

(no-undef)